aws_s3_file_explorer/tasks/s3_file_explorer.airplane.ts

```
@@ -0,0 +1,175 @@
+
+    const res = await client.send(command);
+    if (!res.Contents) {
+      return [];
+    }
+
+    return res.Contents.map((item) => ({
+      name: item.Key,
+      last_modified: item.LastModified,
+      size: item.Size,
+    }));
+  }
+);
+
+export const upload_item = airplane.task(
+  {
+    slug: "upload_item_s3",
+    parameters: {
+      region: "shorttext",
+      bucket_name: "shorttext",
+      file: "upload",
+    },
+    envVars: {
+      AWS_OIDC_ROLE: { config: "aws_oidc_role_test" },
+    },
+    description: "Upload a file to an s3 bucket.",
+  },
+  async (params) => {
+    const client = await newAWSClient(params.region);
+    await params.file.download();
+    const buffer = await params.file.arrayBuffer();
+    const command = new PutObjectCommand({
+      Bucket: params.bucket_name,
+      Key: params.file.name,
+      Body: buffer,
+    });
+
+    client
+      .send(command)
+      .then((result) => {
+        console.log("File uploaded successfully!");
+      })
+      .catch((err) => {
+        console.log("Error uploading file to s3: ", err);
+      });
+
+    await airplane.display.file(params.file);
+    await airplane.display.text("File uploaded successfully!");
+  }
+);
+
+export const download_item = airplane.task(
+  {
+    slug: "download_item_s3",
+    parameters: {
+      region: "shorttext",
+      bucket_name: "shorttext",
+      item_name: "shorttext",
+    },
+    envVars: {
+      AWS_OIDC_ROLE: { config: "aws_oidc_role_test" },
+    },
+  },
+  async (params) => {
+    const client = await newAWSClient(params.region);
+    if (!params.item_name) {
+      return;
+    }
+    console.log("Getting item to display ", params.item_name, "...");
+    const getItemCommand = new GetObjectCommand({
+      Bucket: params.bucket_name,
+      Key: params.item_name,
+    });
+    const item = await client.send(getItemCommand);
+    const itemBody = (await item.Body?.transformToByteArray()) ?? "";
```

throw an error here rather than returning an empty string and then uploading the empty string?